neural_modelling/src/neuron/spike_processing.c

```diff
@@ -163,13 +169,37 @@ void _dma_complete_callback(uint unused, uint tag) {
         do {
             // Are there any more incoming spikes from the same pre-synaptic
             // neuron?
-            subsequent_spikes = in_spikes_is_next_spike_equal(current_buffer->originating_spike);
+            subsequent_spikes = in_spikes_is_next_spike_equal(
+                current_buffer->originating_spike);
 
             // Process synaptic row, writing it back if it's the last time
             // it's going to be processed
-            synapses_process_synaptic_row(time, current_buffer->row,
+            if (!synapses_process_synaptic_row(time, current_buffer->row,
                                           !subsequent_spikes,
-                                          current_buffer_index);
+                                          current_buffer_index)) {
+                log_error(
+                    "Error processing spike 0x%.8x for address 0x%.8x"
+                    "(local=0x%.8x)",
+                    current_buffer->originating_spike,
+                    current_buffer->sdram_writeback_address,
+                    current_buffer->row);
+
+                // Print out the row for debugging
+                for (uint32_t i = 0;
+                        i < (current_buffer->n_bytes_transferred >> 2); i++) {
+                    log_error("%u: 0x%.8x", i, current_buffer->row[i]);
+                }
+
+                // Don't go into real RTE here to allow proper debugging
+                vic[VIC_DISABLE] =
```

Comment:
This was an attempt to avoid the cpu stopping on an error, thus allowing you to read DTCM.  Without this, an RTE stops the core I think, so you can't issue scp commands to that core.  The issue I was trying to debug turned out to be a write issue though, so I don't mind this being replaced if it is preferred.

Sent from my android device.

-----Original Message-----
From: Alan Stokes <notifications@github.com>
To: SpiNNakerManchester/sPyNNaker <sPyNNaker@noreply.github.com>
Cc: Andrew Rowley <Andrew.Rowley@manchester.ac.uk>
Sent: Thu, 30 Jul 2015 17:01
Subject: Re: [sPyNNaker] Speed up data transfer 2 (#112)


In neural_modelling/src/neuron/spike_processing.c<https://github.com/SpiNNakerManchester/sPyNNaker/pull/112#discussion_r35887284>:

> +                                          current_buffer_index)) {
> +                log_error(
> +                    "Error processing spike 0x%.8x for address 0x%.8x"
> +                    "(local=0x%.8x)",
> +                    current_buffer->originating_spike,
> +                    current_buffer->sdram_writeback_address,
> +                    current_buffer->row);
> +
> +                // Print out the row for debugging
> +                for (uint32_t i = 0;
> +                        i < (current_buffer->n_bytes_transferred >> 2); i++) {
> +                    log_error("%u: 0x%.8x", i, current_buffer->row[i]);
> +                }
> +
> +                // Don't go into real RTE here to allow proper debugging
> +                vic[VIC_DISABLE] =


No idea what this bit of code gains you over rt_error(RTE_SWERR)

why are you trying to fake a error during normal runtime behaviour?

—
Reply to this email directly or view it on GitHub<https://github.com/SpiNNakerManchester/sPyNNaker/pull/112/files#r35887284>.